libs/application/templates/new-primary-school/src/lib/dataSchema.ts

```diff
@@ -40,6 +42,34 @@ export const dataSchema = z.object({
     .refine((r) => r === undefined || r.length > 0, {
       params: errorMessages.relativesRequired,
     }),
+  photography: z
+    .object({
+      photographyConsent: z.enum([YES, NO]),
+      photoSchoolPublication: z.string().optional(),
+      photoSocialMediaPublication: z.string().optional(),
+    })
+    .refine(
+      ({ photographyConsent, photoSchoolPublication }) =>
+        photographyConsent === YES
+          ? !!photoSchoolPublication &&
+            [YES, NO].includes(photoSchoolPublication)
+          : true,
+      {
+        path: ['photoSchoolPublication'],
+        params: coreErrorMessages.missingAnswer,
+      },
+    )
+    .refine(
+      ({ photographyConsent, photoSocialMediaPublication }) =>
+        photographyConsent === YES
+          ? !!photoSocialMediaPublication &&
+            [YES, NO].includes(photoSocialMediaPublication)
+          : true,
```

Comment:
Do we need to check if `photoSocialMediaPublication` is YES/NO?

```suggestion
      ({ photographyConsent, photoSocialMediaPublication }) =>
        photographyConsent === YES ? !!photoSocialMediaPublication : true,
```